geopandas/io/tests/test_sql.py

```diff
@@ -473,10 +473,16 @@ def test_write_postgis_without_crs(self, engine_postgis, df_nybb):
                 schema="public", table=table, geom_col="geometry"
             )
         )
+
+        # Testing when replacing table
         with engine.connect() as conn:
             target_srid = conn.execute(sql).fetchone()[0]
         assert target_srid == 0, "SRID should be 0, found %s" % target_srid
 
+        # Testing when append table
+        with pytest.warns(UserWarning, match="Could not parse CRS from the GeoDataF"):
+            write_postgis(df_nybb, con=engine, name=table, if_exists="append")
+
```

Comment:
```suggestion
assert target_srid == 0, "SRID should be 0, found %s" % target_srid

```
You should add the assertion in too, this current just tests that `if_exists="append"` doesn't crash, but we want it to not crash and also give the right answer 😄